app/src/components/market/sections/market_sell/scalar_market_sell.tsx

```diff
@@ -1,12 +1,340 @@
+  const [balanceItem, setBalanceItem] = useState<BalanceItem>(balances[positionIndex])
+  const [status, setStatus] = useState<Status>(Status.Ready)
+  const [message, setMessage] = useState<string>('')
+  const [isModalTransactionResultOpen, setIsModalTransactionResultOpen] = useState(false)
+  const [amountShares, setAmountShares] = useState<Maybe<BigNumber>>(new BigNumber(0))
+  const [amountSharesToDisplay, setAmountSharesToDisplay] = useState<string>('')
+  const [isNegativeAmountShares, setIsNegativeAmountShares] = useState<boolean>(false)
+
+  const lowerBound = scalarLow && Number(formatBigNumber(scalarLow, 18))
+  const upperBound = scalarHigh && Number(formatBigNumber(scalarHigh, 18))
+
+  useEffect(() => {
+    setIsNegativeAmountShares(formatBigNumber(amountShares || Zero, collateral.decimals).includes('-'))
+  }, [amountShares, collateral.decimals])
+
+  useEffect(() => {
+    setBalanceItem(balances[positionIndex])
+    // eslint-disable-next-line
+  }, [balances[positionIndex]])
+
+  useEffect(() => {
+    setPositionIndex(balances[0].shares.gte(balances[1].shares) ? 0 : 1)
+    setBalanceItem(balances[positionIndex])
+    setAmountShares(null)
+    setAmountSharesToDisplay('')
+    // eslint-disable-next-line
+  }, [collateral.address])
+
+  const calcSellAmount = useMemo(
+    () => async (
+      amountShares: BigNumber,
+    ): Promise<[Maybe<BigNumber>, Maybe<number>, Maybe<BigNumber>, Maybe<BigNumber>]> => {
+      const holdings = balances.map(balance => balance.holdings)
+      const holdingsOfSoldOutcome = holdings[positionIndex]
+      const holdingsOfOtherOutcome = holdings.filter((item, index) => {
+        return index !== positionIndex
+      })
+      const marketFeeWithTwoDecimals = Number(formatBigNumber(fee, 18))
+
+      const amountToSell = calcSellAmountInCollateral(
+        // If the transaction incur in some precision error, we need to multiply the amount by some factor, for example  amountShares.mul(99999).div(100000) , bigger the factor, less dust
+        amountShares,
+        holdingsOfSoldOutcome,
+        holdingsOfOtherOutcome,
+        marketFeeWithTwoDecimals,
+      )
+
+      if (!amountToSell) {
+        logger.warn(
+          `Could not compute amount of collateral to sell for '${amountShares.toString()}' and '${holdingsOfSoldOutcome.toString()}'`,
+        )
+        return [null, null, null, null]
+      }
+
+      const balanceAfterTrade = computeBalanceAfterTrade(
+        balances.map(b => b.holdings),
+        positionIndex,
+        amountToSell.mul(-1),
+        amountShares.mul(-1),
+      )
+
+      const pricesAfterTrade = MarketMakerService.getActualPrice(balanceAfterTrade)
+      const potentialValue = mulBN(amountToSell, 1 / (1 - marketFeeWithTwoDecimals))
+      const costFee = potentialValue.sub(amountToSell)
+
+      const newPrediction = pricesAfterTrade[1] * ((upperBound || 0) - (lowerBound || 0)) + (lowerBound || 0)
```

Comment:
@KadenZipfel i was looking at code and it seems that calcSellAmount function for scalar market and in  market_sell is almost identical...could this function maybe be exported and that change accounted for like some extra parameter which handles that small difference?

Reply:
Good catch. I've outsourced it to this issue: https://github.com/protofire/omen-exchange/issues/1429 for the next release to ensure we can get scalar markets out for v1.2.1.